tf.sh

```
@@ -202,6 +202,23 @@ else
     exit 1
 fi
 
+# Define which terraform version to use
+TF_VERSION_FILE=$DIR/stacks/$TERRAFORM_STACK/terraform.version
+echo $TF_VERSION_FILE
+TF_VERSION="0.11"
+if [ -f "$TF_VERSION_FILE" ]; then
+    TF_VERSION=`cat $TF_VERSION_FILE`
+    if [ $TF_VERSION = "0.12" ]; then
+        echo "Working with terraform v0.12 ..."
+        TERRAFORM_BIN=terraform-0.12
+    else
+        echo "Version $TF_VERSION is not supported!"
```

you need to return an error here and exit.

You don't need to TF_VERSION variable anymore
The overall flow should be:
TERRAFORM_BIN="terraform-0.11"

if [ -f "$TF_VERSION_FILE" ]; then
    TF_VERSION=`cat $TF_VERSION_FILE`
    if [ $TF_VERSION = "0.12" ]; then
        echo "Working with terraform v0.12 ..."
        TERRAFORM_BIN=terraform-0.12
   else
       # return an error
       exit 1
   fi
fi